src/libxtp/sternheimer.cc

```diff
@@ -0,0 +1,338 @@
+  this->_mo_coefficients = _orbitals.MOCoefficients();
+  this->_mo_energies = _orbitals.MOEnergies();
+  this->_inverse_overlap = _overlap_Matrix.inverse();
+  this->_Hamiltonian_Matrix = Hamiltonian();
+}    
+
+void Sternheimer::initializeMultishift(int size) { 
+  _multishift.setMatrixSize(size);
+}
+void Sternheimer::initializePade(int size) {
+  _pade.initialize(size);
+  _pade.clear();
+}
+
+Eigen::MatrixXcd Sternheimer::OverlapMatrix() {
+
+  AOBasis basis = _orbitals.SetupDftBasis();
+  AOOverlap overlap;
+  overlap.Fill(basis);
+  return overlap.Matrix();
+}
+
+Eigen::MatrixXcd Sternheimer::DensityMatrix() {
+  return _orbitals.DensityMatrixGroundState();
+}
+
+Eigen::MatrixXcd Sternheimer::Hamiltonian() {
+
+  const Eigen::MatrixXcd& mo_coefficients = _orbitals.MOCoefficients();
+  const Eigen::MatrixXd& mo_energies = _orbitals.MOEnergies().asDiagonal();
+  return _overlap_Matrix * mo_coefficients * mo_energies *
+                      mo_coefficients.transpose() * _overlap_Matrix;
+}
+
+Eigen::MatrixXcd Sternheimer::CoulombMatrix() {
+
+  AOBasis basis = _orbitals.SetupDftBasis();
+  AOCoulomb coulomb;
+  coulomb.Fill(basis);
+  return coulomb.Matrix();
+}
+
+Eigen::MatrixXcd Sternheimer::SternheimerLHS(const Eigen::MatrixXcd& hamiltonian,
+                                             const Eigen::MatrixXcd& inverse_overlap,
+                                             double eps,
+                                             std::complex<double> omega,
+                                             bool pm) const{
+
+  std::complex<double> temp=eps+omega;
+  if (pm != true) {
+    temp = (eps - omega);
+  }
+  Eigen::MatrixXcd LHS=(inverse_overlap*hamiltonian - temp*Eigen::MatrixXcd::Identity(_basis_size,_basis_size));
+//  for(int i=0; i<LHS.cols(); i++){
+//    for(int j=0; j<LHS.rows(); j++){
+//        if(abs(LHS(i,j))<=1e-6){
+//            LHS(i,j)=0.0;   
+//        }
+//    }
+//  }
+  return LHS;
+}
+
+Eigen::VectorXcd Sternheimer::SternheimerRHS(const Eigen::MatrixXcd& inverse_overlap,
+                                             const Eigen::MatrixXcd& density,
+                                             const Eigen::MatrixXcd& pertubation,
+                                             const Eigen::VectorXcd& coeff) const{
+  
+//    std::cout<<"inv overlap"<<std::endl<<inverse_overlap<<std::endl;
+//    std::cout<<"density"<<std::endl<<density<<std::endl;
+//    std::cout<<"pertubation"<<std::endl<<pertubation<<std::endl;
+//    std::cout<<"coeff"<<std::endl<<coeff<<std::endl;
+    Eigen::VectorXcd RHS=-1*(inverse_overlap-density)*pertubation*coeff;
+//    for(int i=0; i<RHS.size(); i++){
+//        if(abs(RHS(i))<=1e-6){
+//            RHS(i)=0.0;   
+//        }
+//    }
+    return RHS;
+}
+
+std::vector<Eigen::MatrixXcd> Sternheimer::DeltaNOneShot(
+    std::vector<std::complex<double>> w,const Eigen::MatrixXd& pertubation) const{
+  
+  double alpha = 2 * (_mo_energies(_num_occ_lvls) - _mo_energies(2));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'alpha' is assigned a value that is never used.](https://app.codacy.com/app/votca-package/xtp/pullRequest?prid=4286864)